src/wrapper.rs

```diff
@@ -123,26 +139,25 @@ macro_rules! glib_wrapper {
 
     (
         $(#[$attr:meta])*
-        pub struct $name:ident(Shared<$ffi_name:path>);
+        pub struct $name:ident(Object<$ffi_name:path>);
 
         match fn {
-            ref => |$ref_arg:ident| $ref_expr:expr,
-            unref => |$unref_arg:ident| $unref_expr:expr,
+            get_type => || $get_type_expr:expr,
         }
     ) => {
-        glib_shared_wrapper!([$($attr)*] $name, $ffi_name, @ref $ref_arg $ref_expr,
-            @unref $unref_arg $unref_expr);
+        glib_object_wrapper!([$($attr)*] $name, $ffi_name, @get_type $get_type_expr, []);
     };
 
     (
         $(#[$attr:meta])*
-        pub struct $name:ident(Object<$ffi_name:path>);
+        pub struct $name:ident(Object<$ffi_name:path>): [$($implements:tt)+];
```

Comment:
If I not wrong again then  `pub struct Bin(Object<ffi::GtkBin>): Container, Widget, Buildable;` not usable now and
example in need be fixed also.